src/nft/hooks/useSubscribeTransactionState.ts

```diff
@@ -0,0 +1,36 @@
+import { BagStatus, TxStateType } from 'nft/types'
+import { useEffect, useRef } from 'react'
+import shallow from 'zustand/shallow'
+
+import { useBag } from './useBag'
+import { useSendTransaction } from './useSendTransaction'
+
+export function useSubscribeTransactionState(setModalIsOpen: (isOpen: boolean) => void) {
+  const transactionState = useSendTransaction((state) => state.state)
+  const setTransactionState = useSendTransaction((state) => state.setState)
+  const transactionStateRef = useRef(transactionState)
+  const { setBagStatus, setLocked: setBagLocked } = useBag(
+    ({ setBagExpanded, setBagStatus, setLocked }) => ({
+      setBagExpanded,
+      setBagStatus,
+      setLocked,
+    }),
+    shallow
+  )
+
+  useEffect(() => {
+    useSendTransaction.subscribe((state) => (transactionStateRef.current = state.state))
+  }, [])
+
+  useEffect(() => {
```

Comment:
Not for now but when the major refactoring goes down, it may be worth exploring if the useEffects can be removed and bagStatus is directly updated only from the hooks in useFetchAssets or from direct user interaction

Reply:
yeah I was thinking about that